x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtStringClaimValidator.java

```diff
@@ -85,4 +107,12 @@ private List<String> getStringClaimValues(FallbackableClaim fallbackableClaim) {
             return fallbackableClaim.getStringListClaimValue();
         }
     }
+
+    private static Predicate<String> predicateFromPatterns(String claimName, Collection<String> patterns) {
+        try {
+            return Automatons.predicate(patterns);
+        } catch (Exception e) {
+            throw new SettingsException("Invalid patterns for allowed claim values for [" + claimName + "].", e);
```

Comment:
Maybe, if we go with also using patterns for more claims in the future. I'm not convinced for subjects only, today, it's warranted, and in any case, it's messy to implement.
This error shows up when the `allowed_subject_patterns` is invalid, and it will show up in the logs as: `"Invalid patterns for allowed claim values for [sub]."` Even if the settings also contain something such as `fallback_claims.sub: client_id` I think the error is telling enough.